### DISCUSSION TITLE
Clin_CT_PUMA

### DISCUSSION THREAD

**Discussion by pdecazes** [ENHANCE-PET/MOOSE#112] [General]
Hello,

Congratulations for your project. 

I would like to use clin_ct_PUMA but have some problems. First of all, the automatic import and installation of the models doesn't work (file incompletely downloaded), so I have to import it manually.

So I first used the link found in the source code
 "url": "https://ucd-emic-muv.s3.us-west-2.amazonaws.com/moose/clin_ct_puma_04042024.zip",

But I got an error during the use
[1/1] Running prediction for S01 using clin_ct_PUMA...Traceback (most recent call last):
 File "/home/piedec/miniconda3/envs/moose-env/bin/moosez", line 8, in <module>
 sys.exit(main())
 File "/home/piedec/miniconda3/envs/moose-env/lib/python3.9/site-packages/moosez/moosez.py", line 198, in main
 predict.predict(model_name, input_dir, output_dir, accelerator)
 File "/home/piedec/miniconda3/envs/moose-env/lib/python3.9/site-packages/moosez/predict.py", line 78, in predict
 postprocess(original_image_files[0], output_dir, model_name)
 File "/home/piedec/miniconda3/envs/moose-env/lib/python3.9/site-packages/moosez/predict.py", line 131, in postprocess
 predicted_image = file_utilities.get_files(output_dir, '.nii.gz')[0]
IndexError: list index out of range

Using the previous file on this link the model works
"url": "https://ucd-emic-muv.s3.us-west-2.amazonaws.com/moose/clin_ct_PUMA_06112023.zip"

Best regards

**Comment by LalithShiyam** [comment]
Hi @pdecazes, thank you for the message. This might be on me. I didn't do a proper release with the updated links. We are facing issues in hosting the model as we are migrating to a new vendor. So it is going to be turbulent for sometime. I am really sorry about that. I made the latest build, kindly check moosez==2.4.5 in a new environment, with the new clin_ct_PUMA. 

Also please use python3.10!

**Comment by mprires** [comment]
Hi @pdecazes I just tested the PUMA model on python3.10 and everything was working as intended. Please let us know if it fixes the problem for you too!

**Comment by pdecazes** [comment]
Effectively, it's OK with python3.10 (both download and running clin_ct_PUMA). Many thanks.

**Comment by LalithShiyam** [comment]
You are welcome - let us know if you have further issues, closing this for now.
